src/lib-components/forms/RadioCards.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+import Uniques from "@/helpers/Uniques"
+import {
+  RadioGroup,
+  RadioGroupDescription,
+  RadioGroupLabel,
+  RadioGroupOption,
+} from "@headlessui/vue"
+import { CheckCircleIcon } from "@heroicons/vue/solid"
+import { computed, ref, useAttrs } from "vue"
+import InputLabel from "./InputLabel.vue"
+import InputHelp from "./InputHelp.vue"
+import FieldsetLegend from "./FieldsetLegend.vue"
+
+/*
+ * NOTE (spk) headless UI introduced a "name" prop that includes a hidden field
+ * to use the modelValue inside of forms.  It does not however resolve the issue of
+ * supporting HTML5 form validation, so we'll add our own hidden radio buttons to support both.
+ *
+ * The headless technique does include supporting complex modelValues such as objects, which we may
+ * need in the future.  We can revist required validation at that time using a singular hidden checkbox.
+ */
+
+type ModelValue = string | number
+
+type RadioCard = {
+  disabled?: boolean
+  help?: string
+  label: string
+  sublabel?: string
+  value: ModelValue
+}
+
+const props = withDefaults(
+  defineProps<{
+    columns?: 2 | 3
+    help?: string
+    legend?: string
+    modelValue?: ModelValue
+    options: RadioCard[]
+  }>(),
+  {
+    columns: undefined,
+    help: "",
+    legend: "",
+    modelValue: undefined,
+  }
+)
+
+const emit = defineEmits<{
+  (e: "update:modelValue", modelValue: ModelValue): void
+}>()
+
+const attrs = useAttrs()
+const uuid = (attrs.id as string) || Uniques.CreateIdAttribute()
+
+// tracking internal state separate from modelValue
+// allows v-model to be undefined by the consumer but still supports
+// the display requirements of the component.
+// this is usful when the component is used inside a form element and
+// tracking v-model isn't required.
+const internalState = ref()
+const invalid = ref<boolean>()
+const checked = computed(() => {
+  if (props.modelValue === undefined) {
+    return internalState.value
+  }
+
+  return props.modelValue
+})
+
+const onChange = (val: ModelValue) => {
+  internalState.value = val
+  invalid.value = false
+  emit("update:modelValue", val)
+}
+
+const nameAttr = computed(() => {
+  return typeof attrs.name === "string" && attrs.name !== "" ? attrs.name : uuid
+})
+</script>
+
+<template>
+  <RadioGroup
+    :modelValue="checked"
+    @update:model-value="onChange"
+    :disabled="typeof attrs.disabled === 'boolean' ? attrs.disabled : false"
+    :aria-invalid="invalid === true ? 'true' : null"
+    :aria-errormessage="invalid === true ? `error-${uuid}` : null"
+  >
+    <RadioGroupLabel v-if="legend" class="block">
```

Comment:
Assuming this markup primarily came from headless/tailwindui so not diving too deep here

Reply:
Yeah, aside from componetizing and html5 required validation support it's very close to a copy paste of tailwind UI wrapping headless.